src/clients/MultiCallerClient.ts

```diff
@@ -41,28 +44,151 @@ export const unknownRevertReasonMethodsToIgnore = new Set([
     this.txnClient = new TransactionClient(logger);
   }
 
-  // Adds all information associated with a transaction to the transaction queue. This is the intention of the
-  // caller to send a transaction. The transaction might not be executable, which should be filtered later.
-  enqueueTransaction(transaction: AugmentedTransaction) {
-    this.transactions.push(transaction);
+  // Adds all information associated with a transaction to the transaction queue.
+  enqueueTransaction(txn: AugmentedTransaction) {
+    if (this.newMulticaller) {
+      // Value transactions are sorted immediately because the UMA multicall implementation rejects them.
+      const txnQueue = txn.value && txn.value.gt(0) ? this.valueTxns : this.txns;
+      if (txnQueue[txn.chainId] === undefined) txnQueue[txn.chainId] = [];
+      txnQueue[txn.chainId].push(txn);
+    } else this.transactions.push(txn);
   }
 
   transactionCount() {
+    if (this.newMulticaller) {
+      return Object.values(this.txns)
+        .concat(Object.values(this.valueTxns))
+        .reduce((count, txnQueue) => (count += txnQueue.length), 0);
+    }
+
     return this.transactions.length;
   }
 
-  clearTransactionQueue() {
-    this.transactions = [];
+  clearTransactionQueue(chainId: number = null) {
+    if (this.newMulticaller) {
+      if (chainId !== null) {
+        this.txns[chainId] = [];
+        this.valueTxns[chainId] = [];
+      } else {
+        this.txns = {};
+        this.valueTxns = {};
+      }
+    } else this.transactions = [];
   }
 
-  async executeTransactionQueue(simulationModeOn = false): Promise<string[]> {
+  async executeTransactionQueue(simulate = false): Promise<string[]> {
+    if (this.newMulticaller) {
+      // For compatibility with the existing implementation, flatten all txn hashes into a single array.
+      // To be resolved once the legacy implementation is removed and the callers have been updated.
+      const txnHashes: { [chainId: number]: string[] } = await this.executeTxnQueues(simulate);
+      return Object.values(txnHashes).flat();
+    }
+
+    return this.executeTxnQueueLegacy(simulate);
+  }
+
+  // For each chain, collate the enqueued transactions and process them in parallel.
+  async executeTxnQueues(simulate = false): Promise<{ [chainId: number]: string[] }> {
+    const chainIds = [...new Set(Object.keys(this.valueTxns).concat(Object.keys(this.txns)))];
+
+    // One promise per chain for parallel execution.
+    const results = await Promise.allSettled(
+      chainIds
+        .filter((_chainId) => {
+          const chainId = Number(_chainId);
+          return this.txns[chainId]?.length > 0 || this.valueTxns[chainId]?.length > 0;
+        })
+        .map((_chainId) => {
+          const chainId = Number(_chainId);
+          const txns: AugmentedTransaction[] = this.txns[chainId];
+          const valueTxns: AugmentedTransaction[] = this.valueTxns[chainId];
```

Comment:
Typescript will let you get away without the undefined, but I think the undefined is more correct if we're trying to be explicit about the types.

```suggestion
          const txns: AugmentedTransaction[] | undefined = this.txns[chainId];
          const valueTxns: AugmentedTransaction[] | undefined = this.valueTxns[chainId];
```

Reply:
Implemented here: https://github.com/across-protocol/relayer-v2/pull/424/commits/0b8a3d0e4e7d22581d11848560575478c0a5620b.